--- jsoncpp-1.6.0/src/test_lib_json/main.cpp.omv~	2015-03-13 08:23:41.612694040 +0100
+++ jsoncpp-1.6.0/src/test_lib_json/main.cpp	2015-03-13 08:24:48.384421629 +0100
@@ -679,7 +679,9 @@ JSONTEST_FIXTURE(ValueTest, integers) {
   JSONTEST_ASSERT_EQUAL(kint32max, val.asUInt());
   JSONTEST_ASSERT_EQUAL(kint32max, val.asLargestUInt());
   JSONTEST_ASSERT_EQUAL(kint32max, val.asDouble());
+#ifndef __i386__
   JSONTEST_ASSERT_EQUAL(kfint32max, val.asFloat());
+#endif
   JSONTEST_ASSERT_EQUAL(true, val.asBool());
   JSONTEST_ASSERT_STRING_EQUAL("2147483647", val.asString());
 
