--- jsoncpp-1.6.0/SConstruct.0001~	2015-03-13 00:06:20.766007671 +0100
+++ jsoncpp-1.6.0/SConstruct	2015-03-13 00:07:03.257493470 +0100
@@ -121,6 +121,12 @@ elif platform.startswith('linux-gcc'):
     env.Tool( 'default' )
     env.Append( LIBS = ['pthread'], CCFLAGS = os.environ.get("CXXFLAGS", "-Wall"), LINKFLAGS=os.environ.get("LDFLAGS", "") )
     env['SHARED_LIB_ENABLED'] = True
+    if 'CXXFLAGS' in os.environ:
+      env.Append( CXXFLAGS=" " + os.environ['CXXFLAGS'] + " " )
+    if 'CPPFLAGS' in os.environ:
+      env.Append( CPPFLAGS=" " + os.environ['CPPFLAGS'] + " " )
+    if 'LINKFLAGS' in os.environ:
+      env.Append( LINKFLAGS=" " + os.environ['LINKFLAGS'] + " " )
 else:
     print "UNSUPPORTED PLATFORM."
     env.Exit(1)
